alibi_detect/utils/pytorch/kernels.py

```diff
@@ -54,30 +125,39 @@ def __init__(
             Whether or not to track gradients w.r.t. `sigma` to allow it to be trained.
         """
         super().__init__()
+        self.parameter_dict['sigma'] = 'bandwidth'
```

Comment:
Perhaps I'm being stupid but I don't see what's going on here?

Reply:
No, you are right for questioning this. 

The original design purpose for this `parameter_dict` is to hold all relevant kernel parameter symbols and names in one place, so we can later implement a generical loop to initialise/train all parameters (i.e. `for params in list(parameter_dict)`)   

However, the current implementation wrote the initialisation for every parameter manually, so this dictionary bit might no longer be necessary.

Reply:
There is now an `infer_kernel_parameter` that does a loop over `parameter_dict`, so the overall behaviour is no longer dependent on the kernel.